### ISSUE TITLE
.JPG files rejected in name ends in numeric char

### ISSUE THREAD

**Issue by jansseng** [OpenShot/openshot-qt#3137]
![image](https://user-images.githubusercontent.com/40327826/71325685-c5891f80-24b5-11ea-9d0e-2c4b09159529.png)

**Describe the bug:**

Drag/Drop .jpg files
If file name ends in Numeric char openshot rejects as invalid.
If file name ends in Alpha char it is accepted.

Example:

COOL0200.jpg rejected
renamed file
COOL0200a.jpg accepted

**System Details:**
- OpenShot Version 2.4.4-dev2
- Operating System  Windows 10

**Screenshots:** (Optional)
![image](https://user-images.githubusercontent.com/40327826/71325685-c5891f80-24b5-11ea-9d0e-2c4b09159529.png)

**Comment by mikelacsa** [comment]
Hi!

It might help the contributors if you can upload your image here?

**Comment by ferdnyc** [comment]
Confirmed, this one's totally my fault. A bug I introduced in #3021.

#3140 will fix it, but as a workaround in the meantime, it turns out the bug is only in `files_treeview.py` — I didn't make the same mistake in `files_listview.py`, which means that right now the two Project Files display modes operate **_slightly_** differently.

But that provides a workaround: For the moment, switching Project Files into "Details View" (via the context menu) will make it able to import images with numeric-ending filenames — the buggy behavior is only present when using "Thumbnail View".

**Comment by ferdnyc** [comment]
Thanks for reporting, @jansseng! It was a stupid, tiny error that only took a couple of minutes to track down and fix, but who knows how long it might've sat there unnoticed if you hadn't brought it to our attention?

**Comment by ferdnyc** [comment]
> For the moment, switching Project Files into "Details View" (via the context menu) will make it able to import images with numeric-ending filenames — the buggy behavior is only present when using "Thumbnail View".

Er, I got those backwards. You **CAN** import files with names like `COOL0200.JPG` when using **_"Thumbnail View"_**. The bug is only present in "Details View".

**Comment by jansseng** [comment]
I’m using windows 10 Pro (all updates) and tried all below formats and all failed with 

 



 

Here is all formats I tried

 



 

BTW…I tried with multiple .jpg files and same result. Renaming and all work

 

Saving same file as .BMP (COOL200.bmp) and it works fine.

 

Merry Christmas and Happy New Year J

 

George

**Comment by jansseng** [comment]
FYI

 

I tried import & Drag/Drop and failed all tries where the file name ended in Numeric char.

 

George

**Comment by ferdnyc** [comment]
@jansseng Your response didn't include what I'm assuming were screenshots, sorry — images attached to email replies don't come through, they have to be posted from GitHub's web interface.

But anyway, I believe I follow what you were saying. Yes, there was definitely an issue with importing files that were named with numbers at the end of the filename part — PR #3140 fixes that in the development code, and is now merged.

Daily Build versions of OpenShot starting from this one should no longer have any issues with importing image files no matter how they're named:
* [OpenShot-v2.4.4-dev2-1577330886-a1158ee2-12a15298-x86_64.exe](https://github.com/OpenShot/openshot-qt/releases/download/daily/OpenShot-v2.4.4-dev2-1577330886-a1158ee2-12a15298-x86_64.exe)

(I've posted that as a GitHub releases link because it hadn't quite shown up on https://openshot.org/download/ when I last checked. But it should be there soon as well, if it isn't already; there's also another new build coming shortly after that, which will also have the image-import fix.

Basically, the issue should be fixed in any Daily Build with a build timestamp — the numeric value that follows the extended version number in the filename — of `1577330886` or greater.)

**Comment by jansseng** [comment]
Thanks

 

Installed Daily Build  <https://github.com/OpenShot/openshot-qt/releases/download/daily/OpenShot-v2.4.4-dev2-1577331825-a1158ee2-12a15298-x86_64.exe> OpenShot-v2.4.4-dev2-1577331825-a1158ee2-12a15298-x86_64.exe (64-bit) (Dec 25, 2019 4:48 PM) and everything worked J J

 

Thanks again

 

George
